### PR TITLE
Back-60

### PR DESCRIPTION
**Pull request by larasodjati**
Created typing event in socket.js that will show users that are actively typing in the chat app.

Notes:
- I copied socket auth Middleware from Manizha's file, to make sure I am using the same file for the user's authentication
- For testing this event in Postman :
    1. Go to **Headers** tab and add a new **Key** named "Authorization" and input the **Value** with "Bearer [token that you 
        manually copy from authentication route]" (no need to use a double quote in this value column)
    2. Go to **Events** tab and add a new event named "typing" and turn on the listen button
    3. Go to the **Message** tab and change it to the JSON file  and on that file type: {"typing": true} to simulate a client typing 
       and {"typing": false} to simulate a client stopped typing, you can change the value true/false one at a time(one by one)
    4. Still in the **Message** tab, add "typing" in the **Ack** column on the right side and mark that column so we just need to 
        press send every time we want to change the message body, connect to the server, and get the data result
    5. The result when a user typing/stops typing can be seen in the vs code terminal
    
 Tips: 
 if you want to see who's the user that typing in each socket window of Postman, you can log in with another email in a different authentication route, and input that token into the new socket request window in Postman(don't add another authorization key in the same socket window). Example as below.
 
image
image
image

### CODE REVIEW

**Comment by Manizha-khorram** [comment]
Well done Lara, this looks great and fun!  I tested this on postman and got the same result.
@jpremo can you please look at this and approve if it is all good from your point of view.?

File: src/controllers/message.js [hunk 1]
```diff
@@ -115,10 +115,35 @@ const deleteMessage = async (req, res) => {
...
+
+        console.log(`${username} is typing...`)
+    } else {
+        // when a user stops typing, server broadcasts a 'typing' event with an empty message to all connected partners
+        socket.broadcast.emit('typing', {
+            username: username,
+            message: '',
+        })
+
+        console.log(`${username} stopped typing.`)
```
Comment (jpremo) [inline src/controllers/message.js:138]: Remove the logs here

File: middleware/authenticationSocket.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    try {
+        const payload = jwt.verify(token, process.env.JWT_SECRET)
+        // attach the user to the socket object
+        socket.user = { userId: payload.userId, username: payload.username }
+        next()
+    } catch (error) {
+        throw new UnauthenticatedError(
+            'Authentication token missing or invalid'
+        )
```
Comment (jpremo) [inline middleware/authenticationSocket.js:23]: This causes an app crash when an expired token is used. I requested changes in @Manizha-khorram's version of this.

**Review by jpremo — APPROVED** [review]
Remove the logs and it should be good to go. The mentioned auth issue can be fixed in another ticket.

**Pull request merged**